app/src/main/java/com/framgia/englishconversation/screen/createComment/CreateCommentViewModel.java

```diff
@@ -0,0 +1,416 @@
+import com.framgia.englishconversation.data.source.callback.DataCallback;
+import com.framgia.englishconversation.data.source.remote.comment.CommentRemoteDataSource;
+import com.framgia.englishconversation.data.source.remote.comment.CommentRepository;
+import com.framgia.englishconversation.record.model.AudioSource;
+import com.framgia.englishconversation.service.BaseStorageService;
+import com.framgia.englishconversation.service.FirebaseUploadService;
+import com.framgia.englishconversation.utils.Constant;
+import com.framgia.englishconversation.utils.FileUtils;
+import com.framgia.englishconversation.utils.navigator.Navigator;
+import com.framgia.englishconversation.widget.dialog.recordingAudio.RecordingAudioBuilder;
+import com.framgia.englishconversation.widget.dialog.recordingAudio.RecordingAudioDialog;
+import com.google.android.exoplayer2.ExoPlayerFactory;
+import com.google.android.exoplayer2.SimpleExoPlayer;
+import com.google.android.exoplayer2.extractor.DefaultExtractorsFactory;
+import com.google.android.exoplayer2.source.ExtractorMediaSource;
+import com.google.android.exoplayer2.trackselection.DefaultTrackSelector;
+import com.google.android.exoplayer2.upstream.DefaultDataSourceFactory;
+import com.google.android.exoplayer2.util.Util;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static com.framgia.englishconversation.service.FirebaseUploadService
+        .ACTION_UPLOAD_MULTI_FILE;
+import static com.framgia.englishconversation.service.FirebaseUploadService.EXTRA_FILES;
+import static com.framgia.englishconversation.service.FirebaseUploadService.EXTRA_FOLDER;
+import static com.framgia.englishconversation.service.FirebaseUploadService.EXTRA_MEDIA_MODEL;
+import static com.framgia.englishconversation.service.FirebaseUploadService.EXTRA_URI;
+
+/**
+ * Exposes the data to be used in the CreateComment screen.
+ */
+
+public class CreateCommentViewModel extends BaseObservable
+        implements CreateCommentContract.ViewModel, PopupMenu.OnMenuItemClickListener {
+
+    private CreateCommentContract.Presenter mPresenter;
+    private String mInputtedComment;
+    private Context mContext;
+    private RecordingAudioDialog mRecordingAudioDialog;
+    private boolean mIsPlaying;
+    private String mTimeInProgressAudio;
+    private Navigator mNavigator;
+    private MultimediaAdapter mMultimediaAdapter;
+    private TimelineModel mTimelineModel;
+    private SimpleExoPlayer mSimpleExoPlayer;
+    private long mCurrentPlaybackPosition = 0;
+    private String mCurrentMultimediaFileUrl;
+    private boolean mIsUploading = false;
+    private BroadcastReceiver mReceiver;
+    private ProgressDialog mProgressDialog;
+    private CommentRepository mCommentRepository;
+
+    public CreateCommentViewModel(Context context, TimelineModel timelineModel) {
+        mContext = context;
+        registerBroadcastReceiver();
```

Comment:
Do you want register BroadcastReceiver in onCreate, I see that you register Broadcast in onStart again, do you ? If you register Broadcast in onCreate(), You must unregister it in onDestroy() of CreateCommentActivity !
If you register Broadcast in onResume(), You must unregister it in onPause() of CreateCommentActivity !
Note: You should not register Broadcast in onStart().
Please visit link: https://developer.android.com/guide/components/broadcasts.html. See Context-registered receivers part !

Reply:
I forgot to delete the register broadcast line in constructor. About the timing of broadcast's registration and unregistration I will still leave it in onStart and onStop. The official documentation does not say anything about the prohibition of register broadcast in onStart. Besides, from Android N split-screen multitasking mode, in order to still update the UI when user is focusing on another screen then you must not register and unregister it onResume and onPause. Nothing wrong with register and unregister broadcast in onStart and onStop, please read the accepted answer in this: https://stackoverflow.com/questions/7887169/android-when-to-register-unregister-broadcast-receivers-created-in-an-activity